File: dsal/lab1/lsearch.c
#include<stdio.h>

int Lsearch(int arr[],int n, int ele);
int main()
{
	int arr[10],i,n,ele,x;
	printf("enter number of elements: \n");
	scanf("%d",&n);
	printf("enter elements: \n");
	
	for(i=0;i<n;i++){
		scanf("%d",&arr[i]);
	}
	printf("enter element to search for: ");
	scanf("%d",&ele);
	x=Lsearch(arr,n,ele);
	if(x)
	{
		printf("element found at position %d",x);
	}
	else
	{
		printf("element not found");
	}
	return 0;
}

int Lsearch(int arr[],int n, int ele)
{
	int i, flag=0;
	for(i=0;i<n;i++){
		if(arr[i]==ele){
			flag=1;
			break;
		}
	}
	if(flag==1)
	{
		return i+1;
	}
	else
	{
		return 0;
	}
}

File: oopl/lab2/lsearch.java
import java.util.Scanner;

class lsearch
{
	
	public static void main(String[] args)
	{
		
		int i,len, key, array[];
		Scanner sc = new Scanner(System.in);
	 	
	 	System.out.println("enter array length:");
	 	len = sc.nextInt(); 
	 	array = new int[len]; 
	 	
	 	System.out.println("enter " + len + " elements:");
	    
	    for (i = 0; i < len; i++)
	        array[i] = sc.nextInt();
	    
	    System.out.println("enter the value to be found:");
	    key = sc.nextInt();
	    
	    for (i = 0; i < len; i++)
	    {
	        if (array[i]== key) 
	        {
	        	System.out.println(key+" is present at position "+(i+1));
	            break;
	        }
	    }
	    
	    if (i == len)
	    	System.out.println(key + " doesn't exist in array");
   }
}